test/createRoutineCreator.js

```diff
@@ -0,0 +1,334 @@
+    expect(routine.REQUEST).to.equal(REQUEST);
+    expect(routine.request.toString()).to.equal(REQUEST);
+    expect(routine.request(payload)).to.deep.equal(requestAction);
+
+    expect(routine.success).to.be.a('function');
+    expect(routine.SUCCESS).to.equal(SUCCESS);
+    expect(routine.success.toString()).to.equal(SUCCESS);
+    expect(routine.success(payload)).to.deep.equal(successAction);
+
+    expect(routine.failure).to.be.a('function');
+    expect(routine.FAILURE).to.equal(FAILURE);
+    expect(routine.failure.toString()).to.equal(FAILURE);
+    expect(routine.failure(payload)).to.deep.equal(failureAction);
+
+    expect(routine.fulfill).to.be.a('function');
+    expect(routine.FULFILL).to.equal(FULFILL);
+    expect(routine.fulfill.toString()).to.equal(FULFILL);
+    expect(routine.fulfill(payload)).to.deep.equal(fulfillAction);
+  });
+
+
+  it('should create an extened routine', () => {
+    const createExtendedRoutine = createRoutineCreator(['SOME_LONG_TYPE', 'OTHER']);
+    const routine = createExtendedRoutine(PREFIX);
+
+    expect(routine).to.be.a('function');
+    expect(routine.toString()).to.equal(TRIGGER);
+    expect(routine(payload)).to.deep.equal(triggerAction);
+
+    expect(routine.trigger).to.be.a('function');
+    expect(routine.TRIGGER).to.equal(TRIGGER);
+    expect(routine.trigger.toString()).to.equal(TRIGGER);
+    expect(routine.trigger(payload)).to.deep.equal(triggerAction);
+
+    expect(routine.request).to.be.a('function');
+    expect(routine.REQUEST).to.equal(REQUEST);
+    expect(routine.request.toString()).to.equal(REQUEST);
+    expect(routine.request(payload)).to.deep.equal(requestAction);
+
+    expect(routine.success).to.be.a('function');
+    expect(routine.SUCCESS).to.equal(SUCCESS);
+    expect(routine.success.toString()).to.equal(SUCCESS);
+    expect(routine.success(payload)).to.deep.equal(successAction);
+
+    expect(routine.failure).to.be.a('function');
+    expect(routine.FAILURE).to.equal(FAILURE);
+    expect(routine.failure.toString()).to.equal(FAILURE);
+    expect(routine.failure(payload)).to.deep.equal(failureAction);
+
+    expect(routine.fulfill).to.be.a('function');
+    expect(routine.FULFILL).to.equal(FULFILL);
+    expect(routine.fulfill.toString()).to.equal(FULFILL);
+    expect(routine.fulfill(payload)).to.deep.equal(fulfillAction);
+
+    expect(routine.someLongType).to.be.a('function');
+    expect(routine.SOME_LONG_TYPE).to.equal(SOME);
+    expect(routine.someLongType.toString()).to.equal(SOME);
+    expect(routine.someLongType(payload)).to.deep.equal(someAction);
+
+    expect(routine.other).to.be.a('function');
+    expect(routine.OTHER).to.equal(OTHER);
+    expect(routine.other.toString()).to.equal(OTHER);
+    expect(routine.other(payload)).to.deep.equal(otherAction);
+  });
+
+  it('should create extened routine if types arg is a string', () => {
+    const createExtendedRoutine = createRoutineCreator('SOME_LONG_TYPE');
```

Comment:
Any examples of stage longer than one word? IMO stage is a short and informative verb: show, hide, open, close, toggle, trigger, request, remove, populate, truncate, ...

Reply:
`TOGGLE_PROJECT_INFO` https://github.com/shapkarin/shapkarin.me/tree/master/src/Pages/Projects

sure, I can name that `TOGGLE` but anyway that `stages` are goes direct to `redux types` and types can be any length.

I imagine that PR feature like I nice way to have all types and actions about some "enity" not only in one namespace but also in one "enity" like yours `routine`. Seems that it makes code more consistent.

Reply:
I added an example with [steps routine](https://github.com/afitiskin/redux-saga-routines/pull/59/files#diff-04c6e90faac2675aa89e2176d2eec7d8R99)